apps/sms/views/inbox/js/inbox.js

```diff
@@ -806,29 +708,16 @@ var InboxView = {
     Threads.delete(threadId);
 
     // Cleanup the DOM
-    this.removeConversationDOM(threadId);
+    var index = this.listData.findIndex((data) => data.id === threadId);
+    if (index > -1) {
+      this.listData.splice(index, 1);
+      this.list.setModel(this.listData);
```

Comment:
I see we do a lot of ```setModel``` calls, IMO it's going to be a pain for the developers, wondering if fast list or we should provide some sort of custom ```Datasource``` or ```ObservableList``` from RxJS wrapper around that, so that all these updates are done a bit more automagically?

```
interface IDataSource<T> {
  raw: T[];
  
  get(index: number): T;
  update(index: number, item: T);
  add(item: T);
  remove(index: number);
  clear();

  // Optional for keyed storage, like threadId
  getByKey(key: string): T;

  on(eventName: string, handler: function);
}
.....
let datasource = new DataSource();

datasource.on('change', Utils.throttle(() => this.list.setModel(datasource.raw)));

```

Maybe not now, but just throwing ideas :)